### PR TITLE
Add cutest_finalize as a finalizer

### LINKED ISSUE #91: `finalizer`
Add a `finalizer` for CUTEst.

Related: https://github.com/JuliaSmoothOptimizers/Optimize.jl/issues/32

### PR DESCRIPTION
**Pull request by abelsiqueira**
Closes #91

### CODE REVIEW

File: test/consistency.jl [hunk 1]
```diff
@@ -15,5 +15,5 @@ for problem in problems
   @printf("Checking problem %-15s%12s\t", problem_s, "")
   consistent_nlps(nlps)
 
-  cutest_finalize(nlp)
+  finalize(nlp_cutest)
```
Comment (dpo) [inline test/consistency.jl]: What's the difference between calling `finalize` and calling `cutest_finalize` directly?

Comment (abelsiqueira) [inline test/consistency.jl]: Hopefully none, but `finalize` is general for Julia, while `cutest_finalize` is very specific. Maybe we shouldn't even export `cutest_finalize` anymore. What do you think?

**Comment by dpo** [comment]
Oops, there's a conflict.

**Comment by abelsiqueira** [comment]
Rebased

**Comment by dpo** [comment]
Thank you!

**Pull request merged**